solutions/communication_kafka/README.md

```diff
@@ -0,0 +1,423 @@
+
+Kafka can be used as a distributed pub/sub messaging system that replaces traditional message brokers like ActiveMQ and RabbitMQ.
+
+### Streams
+
+Kafka can be used to perform real-time computations on event streams.
+
+### Metrics and Monitoring
+
+Kafka can be used to aggregate statistics from distributed applications to produce centralized feeds with real-time metrics.
+
+### Log Aggregation
+
+Kafka can be used as a single source of truth by centralizing logging data across all sources in a distributed system. 
+
+## Kafka Concepts
+
+### Kafka and Events
+
+An event is any type of action, incident, or change that's identified or recorded by software or applications. Kafka models events as key/value pairs. Keys are often primitive types like strings or integers. Values are serialized representations of an application domain object or a raw message.
+
+### Kafka Topics
+
+A topic is a log of events. They are append only: New messages always go to the end of a log. The logs are immutable - once something has happened, it is difficult to make it un-happen.
+Traditional messaging systems have topics and queues, which store messages temporarily for a short time. The advantage of Kafka topics is, that every topic can be configured to expire data after it reaches a certain age, that means the data can be stored for as short as seconds to indefinitely. The logs are files stored on disk.
+
+### Kafka Partitioning 
+
+Partitioning takes the single topic log and breaks in into multiple logs, each of which can live on a separate node in the Kafka cluster. This makes Kafka a distributed system and helps it scale. Kafka messages can be given a key, the key guarantees that all the events from a given entity will always arrive in order. Without a key, the messages will be distributed among all the topic's partitions and any kind of ordering will be ignored.
+
+### Kafka Brokers
+
+Kafka is composed of a network of containers or machines called brokers, each running the Kafka broker process. Each broker hosts some set of partitions and handles incoming requests to write new events to those partitions or read events from them. Brokers also handle replication of partitions between each other.
+
+### Replication
+
+Partition data is copied automatically from a leader broker to follower brokers in the network. If one node in the cluster dies, another will take its role. This helps availability of data.
+
+### Kafka Producers and Consumers
+
+Producers connect and send key/value pairs to specific topic on the cluster. Consumers can then connect to the cluster and subscribe to one or more topics, listening and consuming key/value pairs of Kafka messages. Consuming a message does not destroy it, it can still be read by any other consumer. This differentiates Kafka from other traditional message brokers. Many consumers can read from one topic. Applications can be both consumers and producers.
+
+#### Source: https://developer.confluent.io/what-is-apache-kafka/
```

Comment:
I'm not sure if this is necessary for the solution. Maybe the use cases could be the input for a "Constraints and alternatives" section where you list the use cases (as bullet points) with the hint these are common use cases for kafka.

I would expect for this article that whoever reads this article is ciomfortable in using kafka. If not, he should find here references to good documentation.